desc/optimize/_constraint_wrappers.py

```diff
@@ -1094,3 +1069,48 @@ def constants(self):
     def __getattr__(self, name):
         """For other attributes we defer to the base objective."""
         return getattr(self._objective, name)
+
+
+# in ProximalProjection we have an explicit state that we keep track of (and add
+# to as we go) meaning if we jit anything with self static it doesn't update
+# correctly, while if we leave self unstatic then it recompiles every time because
+# the pytree structure of ProximalProjection is changing. To get around that we
```

Comment:
So, for example ForceBalance constraint changes in terms of value but not the structure so, it doesn't recompile everytime. but do to appended state to the self, structure of ProximalProjection changes? Do I understand correctly?

Reply:
yes exactly. In theory we might be able to get around with with #1034 but that's future work.